app/validators/shelve_book_validator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class ShelveBookValidator < ActiveModel::Validator
```

Comment:
It is not the best practice in rails to define Validation classes like it is in larvel, please define your validations as model functions and call them using validate